src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -1033,6 +1032,21 @@ MavlinkReceiver::handle_message_set_position_target_global_int(mavlink_message_t
 			const double lon = target_global_int.lon_int / 1e7;
 			map_projection_project(&global_local_proj_ref, lat, lon, &setpoint.x, &setpoint.y);
 
+		} else {
+			setpoint.x = NAN;
+			setpoint.y = NAN;
+		}
+
+		// position z
+		if (!(type_mask & (POSITION_TARGET_TYPEMASK_Z_IGNORE))) {
```

Comment:
Currently this is only done for `SET_TARGET_POSITION_GLOBAL_INT`. we need to do the same for `SET_TARGET_POSITION_LOCAL_NED`